Gems/AtomLyIntegration/CommonFeatures/Code/Source/SharedPreview/SharedThumbnailRenderer.cpp

```diff
@@ -51,64 +54,124 @@ namespace AZ
-                    AtomToolsFramework::GetSettingsValue<AZStd::string>(LightingAssetPathSetting, DefaultLightingPresetPath));
+                ThumbnailConfig thumbnailConfig;
+                thumbnailConfig.m_modelAsset.Create(assetInfo.m_assetId);
+                thumbnailConfig.m_materialAsset.Create(SharedPreviewUtils::GetAssetIdForProductPath(
+                    AtomToolsFramework::GetSettingsValue<AZStd::string>(MaterialAssetPathSetting, DefaultMaterialPath)));
+                thumbnailConfig.m_lightingAsset.Create(SharedPreviewUtils::GetAssetIdForProductPath(
+                    AtomToolsFramework::GetSettingsValue<AZStd::string>(LightingAssetPathSetting, DefaultLightingPresetPath)));
+                return thumbnailConfig;
             }
-            else if (assetInfo.m_assetType == RPI::MaterialAsset::RTTI_Type())
+
+            if (assetInfo.m_assetType == RPI::MaterialAsset::RTTI_Type())
             {
                 static constexpr const char* ModelAssetPathSetting =
                     "/O3DE/Atom/CommonFeature/SharedPreview/MaterialAssetType/ModelAssetPath";
                 static constexpr const char* LightingAssetPathSetting =
                     "/O3DE/Atom/CommonFeature/SharedPreview/MaterialAssetType/LightingAssetPath";
 
-                thumbnailConfig.m_modelId = SharedPreviewUtils::GetAssetIdForProductPath(
-                    AtomToolsFramework::GetSettingsValue<AZStd::string>(ModelAssetPathSetting, DefaultModelPath));
-                thumbnailConfig.m_materialId = assetInfo.m_assetId;
-                thumbnailConfig.m_lightingId = SharedPreviewUtils::GetAssetIdForProductPath(
-                    AtomToolsFramework::GetSettingsValue<AZStd::string>(LightingAssetPathSetting, DefaultLightingPresetPath));
+                ThumbnailConfig thumbnailConfig;
+                thumbnailConfig.m_modelAsset.Create(SharedPreviewUtils::GetAssetIdForProductPath(
+                    AtomToolsFramework::GetSettingsValue<AZStd::string>(ModelAssetPathSetting, DefaultModelPath)));
+                thumbnailConfig.m_materialAsset.Create(assetInfo.m_assetId);
+                thumbnailConfig.m_lightingAsset.Create(SharedPreviewUtils::GetAssetIdForProductPath(
+                    AtomToolsFramework::GetSettingsValue<AZStd::string>(LightingAssetPathSetting, DefaultLightingPresetPath)));
+                return thumbnailConfig;
             }
-            else if (assetInfo.m_assetType == RPI::AnyAsset::RTTI_Type())
+
+            if (assetInfo.m_assetType == RPI::MaterialTypeAsset::RTTI_Type())
+            {
+                static constexpr const char* ModelAssetPathSetting =
+                    "/O3DE/Atom/CommonFeature/SharedPreview/MaterialAssetType/ModelAssetPath";
+                static constexpr const char* LightingAssetPathSetting =
+                    "/O3DE/Atom/CommonFeature/SharedPreview/MaterialAssetType/LightingAssetPath";
```

Comment:
Should these be "MaterialTypeAssetType" instead of "MaterialAssetType"? Or factor out these constants with above?